portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +52,45 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String userComment = (String) entity.getProperty("userComment");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        Comment comment = new Comment(id, userComment, timestamp);
+        comments.add(comment);
```

Comment:
minor thing: since you're not reusing the comment object, you can just say

`comments.add(new Comment(id, userComment, timestamp))`